src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/QuestionManager.java

```diff
@@ -633,11 +633,6 @@ public GameFilter createGameFilterForRandomQuestions(final RegisteredUserDTO cur
       List<String> stagesList = new ArrayList<>();
       List<String> examBoardsList = new ArrayList<>();
 
-      for (UserContext uc : userContexts) {
```

Comment:
Removing this entirely means you aren't looking at user contexts at all. So when a user has stage A level and exam board WJEC and doesn't want to see other content, they are still getting GCSE/A level questions from other exam boards.

Reply:
This is also a concern I have. As you mentioned on Slack to me, the if statement on ln.629 counteracts the if statements within this for loop, meaning the code was never being called (tested this too). I agree we need somewhere in the code to take UC into account, maybe one to tackle with Charles, because it is not clear where that is/was.